crates/ark/src/reticulate.rs

```diff
@@ -0,0 +1,145 @@
+use std::sync::Mutex;
+
+use amalthea::comm::comm_channel::CommMsg;
+use amalthea::comm::event::CommManagerEvent;
+use amalthea::socket::comm::CommInitiator;
+use amalthea::socket::comm::CommSocket;
+use anyhow::anyhow;
+use crossbeam::channel::Sender;
+use harp::RObject;
+use lazy_static::lazy_static;
+use libr::R_NilValue;
+use libr::SEXP;
+use serde_json::json;
+use stdext::spawn;
+use stdext::unwrap;
+use uuid::Uuid;
+
+use crate::interface::RMain;
+
+lazy_static! {
+    static ref RETICULATE_COMM_ID: Mutex<Option<String>> = Mutex::new(None);
+}
+
+pub struct ReticulateService {
+    comm: CommSocket,
+    comm_manager_tx: Sender<CommManagerEvent>,
+}
+
+impl ReticulateService {
+    fn start(comm_id: String, comm_manager_tx: Sender<CommManagerEvent>) -> anyhow::Result<String> {
+        let comm = CommSocket::new(
+            CommInitiator::BackEnd,
+            comm_id.clone(),
+            String::from("positron.reticulate"),
+        );
+
+        let service = Self {
+            comm,
+            comm_manager_tx,
+        };
+
+        let event = CommManagerEvent::Opened(service.comm.clone(), serde_json::Value::Null);
+        unwrap!(service.comm_manager_tx.send(event), Err(e) => {
+            log::error!("Reticulate: Could not open comm. Error {e}");
+        });
+
+        spawn!(format!("ark-reticulate-{}", comm_id), move || {
+            unwrap!(service.handle_messages(), Err(err) => {
+                log::error!("Connection Pane: Error while handling messages: {err:?}");
+            });
+        });
+
+        Ok(comm_id)
+    }
+
+    fn handle_messages(&self) -> Result<(), anyhow::Error> {
+        loop {
+            let msg = unwrap!(self.comm.incoming_rx.recv(), Err(err) => {
+                log::error!("Reticulate: Error while receiving message from frontend: {err:?}");
+                break;
+            });
+
+            if let CommMsg::Close = msg {
+                self.comm.outgoing_tx.send(CommMsg::Close).unwrap();
+                break;
+            }
+
+            // Forward data msgs to the frontend
```

Comment:
We're just resending the data messages back? Could you add comments about that please?

Reply:
We don't need at, I copied from another comm by mistake.